dev-docs/design/how-lit-html-works.md

```diff
@@ -0,0 +1,429 @@
+
+All the source code described by this document can be found in [`lit-html.ts`](https://github.com/lit/lit/blob/main/packages/lit-html/src/lit-html.ts).
+
+If anything is unclear or you have questions, reach out on our [Discord channel](https://discord.com/invite/buildWithLit), or ask a question on [GitHub Discussions](https://github.com/lit/lit/discussions). All our communities can be found on the [Lit Community page](https://lit.dev/docs/resources/community/).
+
+# What is lit-html?
+
+lit-html is an HTML templating library. Templates are written in JavaScript by mixing static HTML strings and dynamic JavaScript values using template literals. lit-html enables UI as a function of application state programming model, fast initial rendering, and fast updates that minimally update DOM when state changes.
+
+lit-html is very performant as can be seen in the [JS Frameworks Benchmark](https://krausest.github.io/js-framework-benchmark/). A great companion to this document is [Justin Fagnani's lit-html talk](https://youtu.be/Io6JjgckHbg).
+
+# Foundational Pieces
+
+lit-html's design arises naturally out of combining two powerful web primitives: [JavaScript Tagged Template Literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates) (TTLs) and the HTML [`<template>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) element.
+
+TTLs provide an explicit syntax for separating static strings of HTML content and dynamic JavaScript values. The static strings are immutable and preserve identity across multiple evaluations of the tagged literal making them ideal for use as a cache key.
+
+The `<template>` element provides a container for HTML content that can later be cloned into the document. In a template element: scripts don't run, styles don't apply, custom elements don't upgrade, etc. During rendering, a template's content can be cloned or imported into the Document which makes the cloned nodes active.
+
+# Rendering
+
+## Summary of lit-html rendering phases
+
+lit-html templates are only a description of the UI. They must be rendered with `render()` to affect the DOM. The **Define** and **Render** phases are controlled by the user of Lit, while **Prepare**, **Create**, and **Update** are internal phases.
+
+<ol>
+<li>
+
+**Define**: `` const ui = (title) => html`<h1>${title}</h1>` ``
+
+</li>
+<li>
+
+**Render**: `render(ui("Example Title"), document.body)`
+
+</li>
+<ol type="a">
+<li>
+
+**Prepare**: The first time render is called for each unique TTL, create a `<template>` element containing the static HTML and Lit `TemplatePart`s encoding a mapping for each dynamic JavaScript value to its location in the static HTML.
```

Comment:
The reason we have these internal phrases is that we don't always need to do all of the work, so let's emphasize when each phase does and doesn't run in its description.

I think it's worth calling out here that no matter how many times the `ui` function is called, or how many places into the DOM it's rendered, we'll only run the `Prepare` step for its template once, the first time it's rendered.

Reply:
How might you phrase that call out? Technically this is mentioning that the work is only done once - but I agree that it's not immediately obvious. It's sort of contained in the `"The first time render is called for each unique TTL"` phrase.